pkgs/development/libraries/openmpi/default.nix

```diff
@@ -40,21 +52,28 @@ in stdenv.mkDerivation rec {
     find -name "Makefile.in" -exec sed -i "s/\`date\`/$ts/" \{} \;
   '';
 
-  buildInputs = with stdenv; [ gfortran zlib ]
+  buildInputs = with stdenv; [ gfortran zlib libfabric pmix ucx ]
     ++ lib.optionals isLinux [ libnl numactl ]
     ++ lib.optionals cudaSupport [ cudatoolkit ]
     ++ [ libevent hwloc ]
     ++ lib.optional (isLinux || isFreeBSD) rdma-core;
 
   nativeBuildInputs = [ perl ];
 
-  configureFlags = with stdenv; lib.optional (!cudaSupport) "--disable-mca-dso"
+  configureFlags = with stdenv; [
+    "--with-cma"
+    "--with-hwloc=${hwloc.dev}"
+    "--with-ofi=${libfabric}"
+    "--with-pmi=${pmix}"
+    "--with-pmix=${pmix} --enable-install-libpmix"
+    "--with-verbs=${rdma-core}"
```

Comment:
openmpi is already build against rdma-core. This line is not neccesary.

Reply:
yeah, i've got a bit verbose in some places. will remove that line.